docs/sphinx/examples/python/tutorials/vqe.ipynb

```diff
@@ -5,43 +5,79 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "# Variational Quantum Eigensolver\n",
-    "\n"
+    "# Variational Quantum Eigensolver"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 6,
    "metadata": {},
    "outputs": [],
    "source": [
-    "# !pip install openfermionpyscf"
+    "# !pip install openfermionpyscf matplotlib scipy"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 4,
+   "execution_count": 7,
    "metadata": {},
-   "outputs": [
-    {
-     "name": "stdout",
-     "output_type": "stream",
-     "text": [
-      "-1.1371755939165522\n"
-     ]
-    }
-   ],
+   "outputs": [],
+   "source": [
+    "import cudaq \n",
+    "import matplotlib.pyplot as plt\n",
+    "from scipy.optimize import minimize \n",
+    "import numpy as np \n",
+    "\n",
+    "np.random.seed(42)\n",
+    "\n",
+    "cudaq.set_target(\"nvidia\")\n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let us first begin by defining the molecule and other metadata about the problem."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
    "source": [
-    "import cudaq\n",
+    "n_hydrogens = 2\n",
+    "bond_distance = .7474 # in Angstroms\n",
```

Comment:
I think some light doc comments explaining the rationale behind some of this code would be helpful. The same applies for the rest of the code cells in this file as well. This notebook is pretty code heavy without much instruction